packages/@aws-cdk/aws-ec2-alpha/test/util.test.ts

```diff
@@ -0,0 +1,38 @@
+import { CidrBlock } from '../lib/util';
+
+describe('Tests for the CidrBlock.rangesOverlap method to check if IPv4 ranges overlap', () =>{
+  test('Should return false for non-overlapping IP ranges', () => {
+    const testCidr = new CidrBlock('10.0.0.0/16');
+    const range1 = ['10.0.0.0', '10.0.15.255'] as [string, string];
+    const range2 = ['10.0.128.0', '10.0.143.255'] as [string, string];
+    expect(testCidr.rangesOverlap(range1, range2)).toBe(false);
+  });
+
+  test('Should return true for overlapping IP ranges', () => {
+    const testCidr = new CidrBlock('54.0.0.0/17');
+    const range1 = ['54.0.0.0', '54.0.127.255'] as [string, string];
+    const range2 = ['54.0.100.0', '54.0.192.255'] as [string, string];
+    expect(testCidr.rangesOverlap(range1, range2)).toBe(true);
+  });
+
+  test('Should return true for overlapping IP ranges where one range is completely inside the other', () => {
+    const testCidr = new CidrBlock('10.0.0.0/16');
+    const range1 = ['10.0.0.0', '10.0.127.255'] as [string, string];
+    const range2 = ['10.0.64.0', '10.0.65.255'] as [string, string];
+    expect(testCidr.rangesOverlap(range1, range2)).toBe(true);
+  });
+
+  test('Should return true for overlapping IP ranges where the last IP of one range is the first IP of the other', () => {
+    const testCidr = new CidrBlock('10.0.0.0/16');
+    const range1 = ['10.0.0.0', '10.0.15.255'] as [string, string];
+    const range2 = ['10.0.15.255', '10.0.255.255'] as [string, string];
```

Comment:
Ips from `10.0.15.255` to `10.0.255.255` consists of multiple CIDR range blocks such as below and not a single CIDR range block
```
10.0.15.255/32
10.0.16.0/20
10.0.32.0/19
10.0.64.0/18
10.0.128.0/17
```

nit: It would be better to have ips that are from a single CIDR range block so it will be easy to understand. I would suggest to provide a single CIDR range block something like below.

Range 1: 10.0.0.0/24 [10.0.0.0 - 10.0.0.255]
Range 2: 10.0.0.255/32 [10.0.0.255 - 10.0.0.255]

Note: In a single CIDR range, usually the last ip can't be first ip of another valid CIDR range except for /32. So for this unit test, testing last ip of one range to another range, we could test only for /32 range ip(range 2).

Reply:
The IPs used in some of the unit tests are not valid CIDR ranges, I only added them to test the actual comparison logic in the `rangesOverlap` method. During actual use, the method would not be called with invalid CIDR ranges, but it might worth having tests around the logic regardless, as was recommended in previous feedback.